src/nyc_trees/js/src/lib/storage.js

```diff
@@ -0,0 +1,52 @@
+"use strict";
+
+var $ = require('jquery'),
+    storage = window.localStorage || {};
+
+
+function Storage(options) {
+    var opts = $.extend({}, {
+        key: '',
+        getState: $.noop,
+        validate: $.noop
+    }, options);
+
+    function save() {
+        storage[opts.key] = serialize();
+    }
+
+    function load() {
+        var state = deserialize(storage[opts.key]);
+        if (state) {
+            opts.validate(state);
```

Comment:
I'm in favor of throwing here and letting the consumer decide how to cleanup the mess. I'll have to think about it.